Library/Homebrew/dependency_collector.rb

```diff
@@ -63,11 +65,14 @@ def fetch(spec)
   end
 
   def cache_key(spec)
-    if spec.is_a?(Resource) && spec.download_strategy <= CurlDownloadStrategy
-      File.extname(spec.url)
-    else
-      spec
+    if spec.is_a?(Resource)
+      if spec.download_strategy <= CurlDownloadStrategy
+        return "#{spec.download_strategy}#{File.extname(spec.url).split("?").first}"
+      elsif spec.download_strategy <= VCSDownloadStrategy
```

Comment:
Do we need the condition at all? Would this work?

```suggestion
      else
```

Reply:
It should if it's a safe assumption that any Resource will have a download strategy defined.